query/src/sql/sql_parser.rs

```diff
@@ -73,12 +75,26 @@ impl<'a> DfParser<'a> {
     }
 
     /// Parse a SQL statement and produce a set of statements with dialect
-    pub fn parse_sql(sql: &'a str) -> Result<(Vec<DfStatement<'a>>, Vec<DfHint>), ErrorCode> {
-        let dialect = &GenericDialect {};
-        let start = Instant::now();
-        let result = DfParser::parse_sql_with_dialect(sql, dialect)?;
-        histogram!(super::metrics::METRIC_PARSER_USEDTIME, start.elapsed());
-        Ok(result)
+    pub fn parse_sql(
+        sql: &'a str,
+        typ: SessionType,
```

Comment:
If databend default sql dialect is `MySqlDialect`, the problems is?

Reply:
So we no need a `SessionType` there

Reply:
If use ClickHouse Client connect Databend server or ClickHouse server SQL like 

select “number” will return col value

But use MySQL client it will return a string

Reply:
tests/suites/0_stateless/00_dummy/00_0000_dummy_select_1.sql

like this test

Reply:
Double quota is not sql standard(not allowed by Snowflake, Pg, ClickHouse), it's just for MySQL...

So we need this special `SessionType `